src/main/java/org/testng/DataProviderInformation.java

```diff
@@ -0,0 +1,33 @@
+package org.testng;
+
+import java.lang.reflect.Method;
+
+/**
+ * Represents information pertaining to a data provider that would be invoked.
+ */
+public class DataProviderInformation {
+    private Object instance;
+    private Method method;
+
+    public DataProviderInformation(Method method, Object instance) {
```

Comment:
@juherr - agreed. It was part of my original approach to include that (You can see the comments of #1490 ) Now I am confused as to what to do. @cbeust didn't want it to be included (Please see https://github.com/cbeust/testng/issues/1490#issuecomment-323418116 ).